PodcastGenerator/admin/episodes_ftp_feature.php

```diff
@@ -67,10 +67,22 @@
         $duration = $fileinfo['playtime_string'];           // Get duration
         $bitrate = $fileinfo['audio']['bitrate'];           // Get bitrate
         $frequency = $fileinfo['audio']['sample_rate'];     // Frequency
-        // These are meta tags by the audio file. If not set, use default
-        $title = $fileinfo['tags']['id3v2']['title'][0] != null ? $fileinfo['tags']['id3v2']['title'][0] : pathinfo('../' . $config['upload_dir'] . $new_files[$i], PATHINFO_FILENAME);
-        $comment = $fileinfo['tags']['id3v2']['comment'][0] != null ? $fileinfo['tags']['id3v2']['comment'][0] : "";
-        $author_name = $fileinfo['tags']['id3v2']['artist'][0] != null ? $fileinfo['tags']['id3v2']['artist'][0] : $config['author_name'];
+        // These are meta tags by the audio file. Try id3v2, then id3v1, the default fallbacks.
+        $title = ($fileinfo['tags']['id3v2']['title'][0] != null
+                 ? $fileinfo['tags']['id3v2']['title'][0]
+                 : ($fileinfo['tags']['id3v1']['title'][0] != null
+                    ? $fileinfo['tags']['id3v1']['title'][0]
+                    : pathinfo('../' . $config['upload_dir'] . $new_files[$i], PATHINFO_FILENAME)));
+        $comment = ($fileinfo['tags']['id3v2']['comment'][0] != null
+                   ? $fileinfo['tags']['id3v2']['comment'][0]
+                   : ($fileinfo['tags']['id3v1']['comment'][0] != null
+                      ? $fileinfo['tags']['id3v1']['comment'][0]
+                      : ""));
+        $author_name = ($fileinfo['tags']['id3v2']['artist'][0] != null
+                       ? $fileinfo['tags']['id3v2']['artist'][0]
+                       : ($fileinfo['tags']['id3v1']['artist'][0] != null
+                          ? $fileinfo['tags']['id3v1']['artist'][0]
+                          : $config['author']));
 
```

Comment:
The statements look really long. Why not put this into a function something like this
```
function getid($tag) {
}
```
Where $tag is something like "title", "comment" or "artist".
Maybe add something like the config str as a parameter as well

Reply:
Done.